lua/plenary/path.lua

```diff
@@ -277,6 +279,74 @@ function Path:read()
   return data
 end
 
+function Path:head(lines)
+  lines = lines or 10
+  self = check_self(self)
+  local chunk_size = 256
+
+  local fd = assert(uv.fs_open(self:expand(), "r", 438))
+  local stat = assert(uv.fs_fstat(fd))
+
+  local data = ''
+  local index, count = 0, 0
+  while count < lines and index < stat.size do
+    local read_chunk = assert(uv.fs_read(fd, chunk_size, index))
+
+    local i = 0
+    for char in read_chunk:gmatch"." do
+      if char == '\n' then
+        count = count + 1
+        if count >= lines then break end
+      end
+      index = index + 1
+      i = i + 1
+    end
+    data = data .. read_chunk:sub(1, i)
+  end
+  assert(uv.fs_close(fd))
+
+  -- Remove potential newline at end of file
+  if data:sub(-1) == '\n' then data = data:sub(1, -2) end
+
+  return data
+end
```

Comment:
Works with different `chunk_size`s. Right now i set it to 256. But i also tested 512.

I tried to make a function which handles both, head and tail, like before but tail is just to different. Have to iterate the `chunk` backwards for that.